engine/src/main/java/org/terasology/engine/config/flexible/AutoConfigTypeHandler.java

```diff
@@ -65,7 +65,7 @@ public Optional<AutoConfig> deserialize(PersistedData data) {
             for (Map.Entry<String, PersistedData> entry : data.getAsValueMap().entrySet()) {
                 Field settingField = settingFields.get(entry.getKey());
                 if (settingField == null) {
-                    logger.warn("Cannot to find setting field with name [{}]", entry.getKey());
+                    logger.atWarn().addArgument(() -> entry.getKey()).log("Cannot to find setting field with name [{}]");
```

Comment:
`getKey()` should be only O(1) so impact is neglectable
```suggestion
                    logger.warn("Cannot to find setting field with name [{}]", entry.getKey()); //NOPMD
```